packages/block-library/src/navigation/style.scss

```diff
@@ -171,13 +168,13 @@
 
 	// Custom menu items.
 	// Show submenus on hover.
-	&:hover > .wp-block-navigation-link__container {
+	&:hover > .wp-block-navigation__submenu-container {
 		visibility: visible;
 		opacity: 1;
 	}
 
 	// Keep submenus open when focus is within.
-	&:focus-within > .wp-block-navigation-link__container {
+	&:focus-within > .wp-block-navigation__submenu-container {
 		visibility: visible;
 		opacity: 1;
 	}
```

Comment:
We should now be able to remove the custom page list hover and focus-within rules altogether - the ones between lines 182 and 199.